src/argilla/server/apis/v1/handlers/datasets.py

```diff
@@ -150,15 +160,33 @@ async def list_current_user_dataset_records(
 
     await authorize(current_user, DatasetPolicyV1.get(dataset))
 
-    records = await datasets.list_records_by_dataset_id(
-        db,
-        dataset_id,
-        current_user.id,
-        include=include,
-        response_statuses=response_statuses,
-        offset=offset,
-        limit=limit,
-    )
+    if metadata.metadata_parsed:
+        metadata_filters = await _build_metadata_filters(db, dataset, metadata.metadata_parsed)
+        response_status_filter = await _build_response_status_filter_for_search(response_statuses, user=current_user)
+
+        search_responses = await search_engine.search(
+            dataset=dataset,
+            metadata_filters=metadata_filters,
+            user_response_status_filter=response_status_filter,
+            offset=offset,
+            limit=limit,
+            sort_by=[SortBy(field="inserted_at")],
+        )
+
+        record_ids = [response.record_id for response in search_responses.items]
+        records = await datasets.get_records_by_ids(
+            db=db, dataset_id=dataset_id, record_ids=record_ids, include=include
+        )
```

Comment:
maybe we could move this to a function, so it could be reused in both endpoints

Reply:
Done